src/gui/qgsmaskingwidget.cpp

```diff
@@ -28,29 +28,54 @@
 #include "qgsvectorlayerutils.h"
 #include "qgsmasksymbollayer.h"
 #include "qgsvectorlayerlabeling.h"
+#include "qgsmessagebaritem.h"
 
 QgsMaskingWidget::QgsMaskingWidget( QWidget *parent ) :
   QgsPanelWidget( parent )
 {
   setupUi( this );
+}
+
+void QgsMaskingWidget::onSelectionChanged()
+{
+  // display message if configuration is not consistent
+  bool printMessage =
+    ( mMaskTargetsWidget->selection().empty() && !mMaskSourcesWidget->selection().empty() )
+    || ( !mMaskTargetsWidget->selection().empty() && mMaskSourcesWidget->selection().empty() );
 
-  connect( mMaskTargetsWidget, &QgsSymbolLayerSelectionWidget::changed, this, [&]()
+  if ( mMessageBarItem && !printMessage )
   {
-    mMaskSourcesWidget->setEnabled( ! mMaskTargetsWidget->selection().isEmpty() );
-    emit widgetChanged();
-  } );
-  connect( mMaskSourcesWidget, &QgsMaskSourceSelectionWidget::changed, this, [&]()
+    mMessageBar->popWidget( mMessageBarItem );
+    delete mMessageBarItem;
+    mMessageBarItem = nullptr;
+  }
+  else if ( !mMessageBarItem && printMessage )
   {
-    emit widgetChanged();
-  } );
+    mMessageBarItem = new QgsMessageBarItem( tr( "You need to select both mask sources and masked symbol layers. If not, all configuration will be lost" ), Qgis::Warning, 0, this );
```

Comment:
"If not, all configuration will be lost" => ", otherwise the configuration will be lost" ?

Reply:
This message is way too long for the message bar anyway. It needs to be made much more concise

Reply:
I changed it to "Select both sources and symbol layers or your configuration will be lost"